user/super/com/google/gwt/emul/java/math/BigInteger.java

```diff
@@ -289,27 +315,53 @@ public BigInteger(byte[] val) {
    *           sign is zero and the magnitude contains non-zero entries.
    */
   public BigInteger(int signum, byte[] magnitude) {
+    this(signum, magnitude, 0, magnitude.length);
+  }
+
+  /**
+   * Constructs a new {@code BigInteger} instance with the given sign and the
+   * given magnitude. The sign is given as an integer (-1 for negative, 0 for
+   * zero, 1 for positive). The magnitude is specified as a byte array. The most
+   * significant byte is the entry at index 0.
+   *
+   * @param signum sign of the new {@code BigInteger} (-1 for negative, 0 for
+   *          zero, 1 for positive).
+   * @param magnitude magnitude of the new {@code BigInteger} with the most
+   *          significant byte first.
+   * @param offset the start offset of the binary representation.
+   * @param length the number of bytes to use.
+   * @throws NullPointerException if {@code magnitude == null}.
+   * @throws NumberFormatException if the sign is not one of -1, 0, 1 or if the
+   *           sign is zero and the magnitude contains non-zero entries.
+   */
+  public BigInteger(int signum, byte[] magnitude, int offset, int length) {
     checkNotNull(magnitude);
 
+    if (length < 0 || offset < 0 || magnitude.length < 0 || length > magnitude.length - offset) {
```

Comment:
```suggestion
    if (length < 0 || offset < 0 || length > magnitude.length - offset) {
```
I don't think we need the magnitude length check?